### DISCUSSION TITLE
saving react hook form data to local storage before submitting

### DISCUSSION THREAD

**Discussion by kevinmachstudio** [react-hook-form/react-hook-form#8899] [Q&A]
is there a way to save the form data to the browser's local storage (before submitting)?

**Comment by Moshyfawn** [comment]
Do whatever logic you want onSubmit, like saving the data to local storage

**Reply by kevinmachstudio** [reply]
if I'm understanding correctly onSubmit happens after submitting, is there a way to do it before any submitting happens?

**Reply by Moshyfawn** [reply]
`onSubmit` is your submit logic.. `handleSubmit` is called with all your validation logic and spits out the `onSubmit`event and the parsed form data into the `onSubmit` for you to `post` it or do whatever.

If you want to run some logic before the validation happens _(provided `mode: "onSubmit"`)_, you can theoretically do
```js
const { getValues } = useForm()

const doCustomStuffToData = event => {
  ...
  const data = getValues()
  saveToLocalStorage(data)
  ...
}

onSubmit={doCustomStuffToData(handleSubmit(onSubmit))}
```

But I don't see my you can't store your data into `localStorage` inside `onSubmit`
